core/src/main/java/io/confluent/kafka/schemaregistry/storage/InMemoryCache.java

```diff
@@ -174,26 +176,53 @@ public Mode mode(String subject,
   }
 
   @Override
-  public Set<String> subjects(String namespace) throws StoreException {
-    Iterator<K> allKeys = getAllKeys();
-    return findSubject(allKeys, namespace);
+  public Set<String> subjects(String subject) throws StoreException {
+    return subjects(matchingPredicate(subject));
   }
 
-  private Set<String> findSubject(Iterator<K> allKeys, String subject) {
-    Set<String> subjects = new HashSet<String>();
-    while (allKeys.hasNext()) {
-      K k = allKeys.next();
+  @Override
+  public Set<String> subjects(Predicate<String> match) throws StoreException {
+    return store.entrySet().stream()
+        .flatMap(e -> {
+          K k = e.getKey();
+          V v = e.getValue();
+          if (k instanceof SchemaKey) {
+            SchemaKey key = (SchemaKey) k;
+            SchemaValue value = (SchemaValue) v;
+            if (value != null && !value.isDeleted()) {
+              if (match.test(key.getSubject())) {
+                return Stream.of(key.getSubject());
```

Comment:
Based on this implementation, should this method return a boolean,  perhaps called `hasSubjects` or something along those lines?